builtin/logical/mysql/path_role_create.go

```diff
@@ -51,9 +52,15 @@ func (b *backend) pathRoleCreateRead(
 	}
 
 	// Generate our username and password. MySQL limits user to 16 characters
-	displayName := req.DisplayName
-	if len(displayName) > 10 {
-		displayName = displayName[:10]
+	displayName := name
```

Comment:
@jippi, making UsernameLength apply for the 'final' username should resolve this. https://github.com/hashicorp/vault/pull/1604/files#r70671797